ros/angel_system_nodes/angel_system_nodes/object_descriptor_generator.py

```diff
@@ -0,0 +1,215 @@
+        self._feature_layer = self.declare_parameter("feature_layer", "head").get_parameter_value().string_value
+
+        # Currently supported layers from torch's resnet_fpn_backbone
+        allowable_layers = ['0', '1', '2', '3', 'pool', 'head']
+        if self._feature_layer not in allowable_layers:
+            raise ValueError(
+                f"Layer {self._feature_layer} not currently supported"
+            )
+
+        log = self.get_logger()
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Torch device? {self._torch_device}")
+        log.info(f"Feature layer? {self._feature_layer}")
+
+        self._model = None
+
+        self._subscription = self.create_subscription(
+            Image,
+            self._image_topic,
+            self.listener_callback,
+            1
+        )
+        self._publisher = self.create_publisher(
+            ObjectDescriptors,
+            self._desc_topic,
+            1
+        )
+
+        self.transforms = transforms.Compose([
+            transforms.ToTensor(),
+        ])
+
+    def get_model(self) -> "torch.nn.Module":
+        """
+        Lazy load the torch model in an idempotent manner.
+        :raises RuntimeError: Use of CUDA was requested but is not available.
+        """
+        model = self._model
+        if model is None:
+            if self._feature_layer == "head":
+                # Use default resnet from torch.models
+                model = torchvision.models.detection.fasterrcnn_resnet50_fpn(
+                    pretrained=True,
+                    progress=False,
+                )
+            else:
+                # Use custom resnet that overrides the forward method call
+                # so that we can access the output of the other layers
+                model = ResNetDescriptors()
+
+            model = model.eval()
+
+            # Transfer the model to the requested device
+            if self._torch_device != 'cpu':
+                if torch.cuda.is_available():
+                    model_device = torch.device(device=self._torch_device)
+                    model = model.to(device=model_device)
+                else:
+                    raise RuntimeError(
+                        "Use of CUDA requested but not available."
+                    )
+            else:
+                model_device = torch.device(self._torch_device)
+
+            self._model = model
+            self._model_device = model_device
+
+        return model
+
+    def listener_callback(self, image):
+        log = self.get_logger()
+
+        model = self.get_model()
+
+        # Convert ROS Image message to CV2
+        rgb_image = BRIDGE.imgmsg_to_cv2(image, desired_encoding="rgb8")
+
+        # Convert to tensor
+        image_tensor = self.transforms(rgb_image)
+        image_tensor = image_tensor.unsqueeze(0)
+
+        # Send to model
+        output = model(image_tensor)
```

Comment:
I presume we want no-grad around this (or decorated on this method)
```suggestion
        with torch.no_grad():
            output = model(image_tensor)
```